x-pack/plugins/ml/server/routes/alerting.ts

```diff
@@ -31,11 +33,11 @@ export function alertingRoutes({ router, routeGuard }: RouteInitialization) {
         tags: ['access:ml:canGetJobs'],
       },
     },
-    routeGuard.fullLicenseAPIGuard(async ({ mlClient, request, response, client }) => {
+    routeGuard.fullLicenseAPIGuard(async ({ mlClient, request, response, client, context }) => {
       try {
-        const alertingService = alertingServiceProvider(
-          mlClient,
-          datafeedsProvider(client, mlClient)
+        const alertingService = sharedServicesProviders.alertingServiceProvider(
+          context.core.savedObjects.client,
```

Comment:
Curious why this is being switched from the mlClient?

Reply:
@alvarezmelissa87 because I need an instance of `alertingServiceProvider` that already takes care of dependencies like `FieldFormatsRegistryProvider `

Reply:
`mlClient` provides space awareness for jobs, we should be using it for all calls to ml endpoints in es.
If alerting is not calling ml endpoints this change is ok, however the definition of `alertingServiceProvider` still expects a `MlClient`, so that will need to be changed.

Reply:
@jgowdyelastic it already uses the`mlClient`, you can find it [here](https://github.com/elastic/kibana/blob/2d4bbd3de0b2fb79ecf8539882d408f937e04e4e/x-pack/plugins/ml/server/shared_services/providers/alerting_service.ts#L13)

Reply:
The function being called here is:
https://github.com/elastic/kibana/blob/2d4bbd3de0b2fb79ecf8539882d408f937e04e4e/x-pack/plugins/ml/server/shared_services/providers/alerting_service.ts#L13

Which takes an `MlClient`

Reply:
ok apologies, i read `context.core.savedObjects.client` as being the es client, not the saved object client.

Reply:
`mlClient` and `client` are used in this function now and can be removed.